Text::Xslate 3.5.0 and 3.5.5 do not build on CentOS 5 (Perl 5.8.8)

I get this output when trying to build the current version of Text::Xslate on CentOS 5, which ships with Perl 5.8.8:

<pre>
Checking dependencies from MYMETA.json ...
Checking if you have Test::Requires 0 ... Yes (0.10)
Checking if you have Scalar::Util 1.14 ... Yes (1.49)
Checking if you have parent 0.221 ... Yes (0.236)
Checking if you have Storable 2.15 ... Yes (2.15)
Checking if you have Encode 2.26 ... Yes (2.93)
Checking if you have Test::More 0.98 ... Yes (1.302120)
Checking if you have Mouse v2.5.0 ... Yes (v2.5.1)
Checking if you have File::Path 2.07 ... Yes (2.15)
Checking if you have File::Copy::Recursive 0 ... Yes (0.39)
Checking if you have Data::MessagePack 0.38 ... Yes (1.00)
Checking if you have ExtUtils::CBuilder 0 ... Yes (0.280230)
Building and testing Text-Xslate-v3.5.5
Building Text-Xslate
/usr/bin/perl tool/opcode_for_pp.PL src/xslate_opcode.inc > lib/Text/Xslate/PP/C
onst.pm
/usr/bin/perl tool/opcode.PL src/xslate_opcode.inc > src/xslate_ops.h
gcc -Isrc -I/usr/lib64/perl5/5.8.8/x86_64-linux-thread-multi/CORE -fPIC -Wall -W
extra -Wdeclaration-after-statement -c -D_REENTRANT -D_GNU_SOURCE -fno-strict-al
iasing -pipe -Wdeclaration-after-statement -I/usr/local/include -D_LARGEFILE_SOU
RCE -D_FILE_OFFSET_BITS=64 -I/usr/include/gdbm -O2 -g -pipe -Wall -Wp,-D_FORTIFY
_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=g
eneric -o src/xslate_methods.o src/xslate_methods.c
In file included from src/xshelper.h:44,
                 from src/xslate.h:3,
                 from src/xslate_methods.xs:1:
src/ppport.h: In function ‘DPPP_my_gv_fetchpvn_flags’:
src/ppport.h:5987: warning: unused parameter ‘flags’
src/ppport.h:5987: warning: unused parameter ‘types’
src/xslate_methods.xs: In function ‘tx_sv_cmp’:
src/xslate_methods.xs:137: error: expected ‘)’ before ‘XS_VERSION’
src/xslate_methods.xs:137: error: too few arguments to function ‘Perl_hv_fetch’
src/xslate_methods.xs: In function ‘tx_prepare_compare_func’:
src/xslate_methods.xs:162: error: expected ‘)’ before ‘XS_VERSION’
src/xslate_methods.xs:162: error: too few arguments to function ‘Perl_hv_fetch’
src/xslate_methods.xs: In function ‘tx_bm_hash_kv’:
src/xslate_methods.xs:352: error: expected ‘)’ before ‘XS_VERSION’
src/xslate_methods.xs:352: error: too few arguments to function ‘Perl_hv_fetch’
src/xslate_methods.xs: In function ‘XS_Text__Xslate__Type__Pair_CLONE’:
src/xslate_methods.xs:529: error: expected ‘)’ before ‘XS_VERSION’
src/xslate_methods.xs:529: error: too few arguments to function ‘Perl_hv_fetch’
src/xslate_methods.xs: In function ‘boot_Text__Xslate__Methods’:
src/xslate_methods.xs:519: error: expected ‘)’ before ‘XS_VERSION’
src/xslate_methods.xs:519: error: too few arguments to function ‘Perl_hv_fetch’
error building src/xslate_methods.o from 'src/xslate_methods.c' at /usr/lib/perl
5/site_perl/5.8.8/ExtUtils/CBuilder/Base.pm line 184, <DATA> line 1.
-> FAIL Installing Text::Xslate failed. See /home/etr/.cpanm/work/1515387677.592
/build.log for details. Retry with --force to force install it.</pre>

I tried forcing installation of the oldest version still available on the CPAN mirrors, 3.5.0, with the same result.

I have in the past been able to build 3.3.9 and 3.4.0 on these systems, and I have tarballs for these locally, so I have a workaround, but I'd prefer to be installing from cpanm.

--- a/.gitignore
+++ b/.gitignore
@@ -35,3 +35,4 @@ lib/Text/Xslate/PP/Const.pm
 /Build
 /_build
 /lib/Text/Xslate.xs
+/src/xs_version.h

--- a/builder/MyBuilder.pm
+++ b/builder/MyBuilder.pm
@@ -23,6 +23,14 @@ sub new {
     $self;
 }
 
+sub _write_xs_version {
+    my ($self, $file) = @_;
+    open my $fh, ">", $file or die "$file: $!";
+    print  {$fh} "#ifndef XS_VERSION\n";
+    printf {$fh} "#  define XS_VERSION \"%s\"\n", $self->dist_version;
+    print  {$fh} "#endif\n";
+}
+
 sub _derive_opcode {
     my ($self, $script, $source, $derived) = @_;
     my @cmd = ($self->{properties}{perl}, $script, $source);
@@ -33,6 +41,10 @@ sub _derive_opcode {
 
 sub ACTION_code {
     my ($self, @args) = @_;
+
+    if (!$self->pureperl_only) {
+        $self->_write_xs_version("src/xs_version.h");
+    }
 
     my @derive = (
         {

--- a/src/xslate_methods.xs
+++ b/src/xslate_methods.xs
@@ -1,3 +1,4 @@
+#include "xs_version.h"
 #include "xslate.h"
 
 #define TXBM_DECL(name) void name \

